modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductSubscriptions.testcase

```diff
@@ -204,6 +204,174 @@ definition {
+
+			CommerceEntry.gotoMenuTab(menuTab = "Subscription");
+		}
+
+		task ("When the Payment Subscription is enabled") {
+			Click(locator1 = "CommerceEntry#PAYMENT_SUBSCRIPTION_TOGGLE");
+		}
+
+		task ("Then the default settings(Type:Day,Length:1,Type in Length:Day) are displayed") {
+			AssertSelectedLabel(
+				key_subscriptionFieldLabel = "Subscription Type",
+				key_subscriptionName = "Payment Subscription",
+				locator1 = "CommerceEntry#SUBSCRIPTION_FIELD_LABEL",
+				value1 = "Day");
+
+			AssertTextEquals(
+				key_fieldLabel = "Subscription Length",
+				key_indexNumber = 1,
+				locator1 = "TextInput#GENERIC_TEXT_INPUT_N",
+				value1 = 1);
+
+			AssertTextEquals(
+				key_subscriptionFieldLabel = "Subscription Length",
+				key_subscriptionName = "Payment Subscription",
+				locator1 = "CommerceEntry#SUBSCRIPTION_TYPE_IN_LENGTH",
+				value1 = "Day");
+		}
+
+		task ("When switching all the subscription type fields when 'Subscription Length = 1'") {
+			for (var subscriptionType : list "Week,Month,Year") {
+				Select(
+					key_subscriptionFieldLabel = "Subscription Type",
+					key_subscriptionName = "Payment Subscription",
+					locator1 = "CommerceEntry#SUBSCRIPTION_FIELD_LABEL",
+					value1 = ${subscriptionType});
+
+				Button.clickPublish();
+
+				task ("Then the subscription type field in Subscription Length is updated accordingly and is singular") {
+					AssertTextEquals(
+						key_subscriptionFieldLabel = "Subscription Length",
+						key_subscriptionName = "Payment Subscription",
+						locator1 = "CommerceEntry#SUBSCRIPTION_TYPE_IN_LENGTH",
+						value1 = ${subscriptionType});
+				}
+			}
+		}
+
+		task ("But if switching all the subscription type fields when 'Subscription Length > 1'") {
+			Type(
+				key_fieldLabel = "Subscription Length",
+				key_indexNumber = 1,
+				locator1 = "TextInput#GENERIC_TEXT_INPUT_N",
+				value1 = 2);
+
+			for (var subscriptionTypeSet : list "Day:Days,Week:Weeks,Month:Months,Year:Years") {
+				var subscriptionType = StringUtil.extractFirst(${subscriptionTypeSet}, ":");
+				var subscriptionTypeInLength = StringUtil.extractLast(${subscriptionTypeSet}, ":");
+
+				Select(
+					key_subscriptionFieldLabel = "Subscription Type",
+					key_subscriptionName = "Payment Subscription",
+					locator1 = "CommerceEntry#SUBSCRIPTION_FIELD_LABEL",
+					value1 = ${subscriptionType});
+
+				Button.clickPublish();
+
+				task ("Then the subscription type field in Subscription Length is updated accordingly and is plural") {
+					AssertTextEquals(
+						key_subscriptionFieldLabel = "Subscription Length",
+						key_subscriptionName = "Payment Subscription",
+						locator1 = "CommerceEntry#SUBSCRIPTION_TYPE_IN_LENGTH",
+						value1 = ${subscriptionTypeInLength});
+				}
+			}
+
+			Click(locator1 = "CommerceEntry#PAYMENT_SUBSCRIPTION_TOGGLE");
+
+			Button.clickPublish();
+		}
+
+		task ("When the Delivery Subscription is enabled") {
```

Comment:
Hi Jayda I think it would be better to have 2 tests instead of 1.
One for payment and the other for delivery subscription.  I think this one is too long.

Reply:
Okay, thanks Luca.